lib/symlink.go

```diff
@@ -89,9 +89,9 @@ func CheckSymlink(symlinkPath string) bool {
 }
 
 // ChangeSymlink : move symlink to existing binary
-func ChangeSymlink(binVersionPath string, binPath string) {
+func ChangeSymlink(product Product, binVersionPath string, binPath string) {
```

Comment:
This breaks a public method

Reply:
As I already commented above, maybe have a fallback for Product to provide backward compatibility? 🤔

Reply:
Since this is not a simple string argument and is taking the "product" (similar to a lot of other public methods that we've created deprecated stubs for) - I wonder if it's better to do the same here.
I'll do so, post the commit and see what you think :)

Reply:
> I've modified in [d22a046](https://github.com/warrensbox/terraform-switcher/commit/d22a046234da91c69f6acdc4d9a5bcd2a8f530cc) - what you think :)

The "trick" looks good =) Maybe also add to comment that this method is still there for backward compatibility as it's public one? 🤔

Reply:
Different to:
```
// This function has been deprecated in favor of ChangeProductSymlink
```
?